web/themes/custom/usagov/usagov.theme

```diff
@@ -10,13 +10,158 @@ use Drupal\taxonomy\Entity\Term;
+  $numH2s = count($dom->getElementsByTagName('h2'));
+
+  if ($numH2s > 1) {
+    // There are multiple H2s
+    // Each H2 is a question
+    $question = "No Questions Found Yet";
+
+    //loop through elements of the body content
+    foreach ($elements as $element) {
+      // Check the type of element
+      if ($element->nodeName == "h2") {
+        // $element is a new question
+        // If there was an existing question, store it and the answer
+        if ($question != "No Questions Found Yet") {
+          $answer = faq_answer_inner_html($answer);
+          $faq[] = (object) ['question' => $question, 'answer' => $answer];
+        }
+
+        // Set up the new question
+        $question = $element->nodeValue;
+
+        // Set up the new answer
+        $answer = new DomDocument();
+        $answer->loadHTML("<html><body></body></html>");
+
+      }
+      elseif ($element->nodeName == "drupal-paragraph") {
+        // $element is an embedded paragraph
+        faq_process_paragraph($element, $answer);
+      }
+      elseif ($element->nodeName == "#text") {
+        // $element is a text node
+        // This is white space that we can ignore.
+      }
+      else {
+        // $element is some kind of content
+        // Append the content to the current answer
+        $importedNode = $answer->importNode($element, TRUE);
+        $answer->documentElement->childNodes[0]->appendChild($importedNode);
+      }
+    }
+
+    // If there was an existing question, store it and the answer
+    if ($question != "No Questions Found Yet") {
+      $answer = faq_answer_inner_html($answer);
+      $faq[] = (object) ['question' => $question, 'answer' => $answer];
+    }
+
+  }
+  else {
+    // There are not multiple H2s
+    // The H1 is the question
+    $question = $node->getTitle();
+
+    // Setup the answer
+    $answer->loadHTML("<html><body></body></html>");
+
+    // Append the intro to the answer
+    $introText = $node->get('field_page_intro')[0]->value;
+    $intro = $answer->createElement('p');
+    $intro->textContent = $introText;
+    $answer->documentElement->childNodes[0]->appendChild($intro);
+
+    //loop through elements of the body content
+    foreach ($elements as $element) {
+      // Check the type of element
+      if ($element->nodeName == "drupal-paragraph") {
+        // $element is an embedded paragraph
+        faq_process_paragraph($element, $answer);
+      }
+      elseif ($element->nodeName == "#text") {
+        // $element is a text node
+        // This is white space that we can ignore.
+      }
+      else {
+        // $element is content for the answer
+        // Append the content to the current answer
+        $importedNode = $answer->importNode($element, TRUE);
+        $answer->documentElement->childNodes[0]->appendChild($importedNode);
+      }
+    }
+
+    $answer = faq_answer_inner_html($answer);
+    $faq[] = (object) ['question' => $question, 'answer' => $answer];
```

Comment:
It's probably a good idea to check `$question` and `$answer` for validity here too, before populating `$faq`.